test/test_prototype_transforms_functional.py

```diff
@@ -1168,25 +1157,67 @@ def _compute_expected_bbox(bbox, padding_):
         torch.testing.assert_close(output_boxes, expected_bboxes)
 
 
+@pytest.mark.parametrize("device", cpu_and_gpu())
+def test_correctness_pad_segmentation_mask_on_fixed_input(device):
+    mask = torch.ones((1, 3, 3), dtype=torch.long, device=device)
+
+    out_mask = F.pad_segmentation_mask(mask, padding=[1, 1, 1, 1])
+
+    expected_mask = torch.zeros((1, 5, 5), dtype=torch.long, device=device)
+    expected_mask[:, 1:-1, 1:-1] = 1
+    torch.testing.assert_close(out_mask, expected_mask)
+
+
 @pytest.mark.parametrize("padding", [[1, 2, 3, 4], [1], 1, [1, 2]])
-def test_correctness_pad_segmentation_mask(padding):
-    def _compute_expected_mask(mask, padding_):
+@pytest.mark.parametrize("padding_mode", ["constant", "edge", "reflect", "symmetric"])
+def test_correctness_pad_segmentation_mask(padding, padding_mode):
+    def _compute_expected_mask(mask, padding_, padding_mode_):
         h, w = mask.shape[-2], mask.shape[-1]
         pad_left, pad_up, pad_right, pad_down = _parse_padding(padding_)
 
+        # expected output can be computed on positive pad values only
+        # but F.pad_* can also crop for negative values
+        assert pad_left > 0 and pad_up > 0 and pad_right > 0 and pad_down > 0
+
         new_h = h + pad_up + pad_down
         new_w = w + pad_left + pad_right
 
         new_shape = (*mask.shape[:-2], new_h, new_w) if len(mask.shape) > 2 else (new_h, new_w)
-        expected_mask = torch.zeros(new_shape, dtype=torch.long)
-        expected_mask[..., pad_up:-pad_down, pad_left:-pad_right] = mask
+        output = torch.zeros(new_shape, dtype=mask.dtype)
+        output[..., pad_up:-pad_down, pad_left:-pad_right] = mask
+
+        if padding_mode_ == "edge":
+            output[..., :pad_up, :pad_left] = mask[..., 0, 0].unsqueeze(-1).unsqueeze(-2)
+            output[..., pad_up:-pad_down, :pad_left] = mask[..., :, 0].unsqueeze(-1)
+            output[..., -pad_down:, :pad_left] = mask[..., -1, 0].unsqueeze(-1).unsqueeze(-2)
+
+            output[..., :pad_up, -pad_right:] = mask[..., 0, -1].unsqueeze(-1).unsqueeze(-2)
+            output[..., pad_up:-pad_down, -pad_right:] = mask[..., :, -1].unsqueeze(-1)
+            output[..., -pad_down:, -pad_right:] = mask[..., -1, -1].unsqueeze(-1).unsqueeze(-2)
+
+            output[..., :pad_up, pad_left:-pad_right] = mask[..., 0, :].unsqueeze(-2)
+            output[..., -pad_down:, pad_left:-pad_right] = mask[..., -1, :].unsqueeze(-2)
```

Comment:
This is quite hard to parse, but I guess there is little we can do to improve. Two suggestions though:

1. If you specify a range as index, the dimensions will not be squeezed. For example: `mask[..., 0, 0].unsqueeze(-1).unsqueeze(-2) == mask[..., :1, :1]`
2. Could you add a comment above each block which side, i.e. top, bottom, left, or right, the padding applies to? Please also do so in the branch below.

Reply:
Let's keep `unsqueeze`, it is much less obvious to parse with `mask[..., :1, :1]`.